### PR TITLE
Adds NWN_SQL_PreparedNULL.

### PR DESCRIPTION
**Pull request by drake127**
I added a function to prepare NULL parameters to the SQL. I tested it with MySQL but not with Postgres or SQLite (but it's not difficult change there). If somebody is willing to test it, it would be great.

### CODE REVIEW

**Comment by drake127** [comment]
Is there anything I can do better to have it merged?

I understand we don't handle NULLs properly on reading but this is useful for SQL (foreign) key checks.

**Comment by mtijanic** [comment]
Hi, I'm sorry you are waiting this long, but I (and I guess other maintainers) don't have time to look at this in detail right now. This weekend hopefully!

Normally, new functions are not given much scrutiny when the worst case is they don't workand nothingis lost. But this one actually touches all of the existing SQL stuff, so it would need a close look, and ideally tests on all three targets, and that's not something we can always do promptly. Sorry!

**Comment by drake127** [comment]
No worries, I am trying to contribute upstream because it gives us advantage of easy upgrade process. Take your time a let me know if there's anything I can improve.

**Comment by drake127** [comment]
I added WebHooks fix to this branch by accident. Should I split it, or you don't mind?

**Comment by mtijanic** [comment]
We don't mind too much if it's the same PR, so long as the commits are different. But again a nom-obvious change will require a more thorough look.

File: Plugins/WebHook/External/httplib.h [hunk 1]
```diff
@@ -441,20 +441,12 @@ inline int select_read(socket_t sock, size_t sec, size_t usec)
...
-    tv.tv_usec = usec;
-
-    if (select(sock + 1, &fdsr, &fdsw, &fdse, &tv) < 0) {
+    pollfd pfd;
+    pfd.fd = sock;
+    pfd.events = POLLIN | POLLOUT;
+    if (poll(&pfd, 1, sec * 1000 + usec / 1000) < 0) {
         return false;
-    } else if (FD_ISSET(sock, &fdsr) || FD_ISSET(sock, &fdsw)) {
+    } else if (pfd.revents & (POLLIN | POLLOUT)) {
```
Comment (mtijanic) [inline Plugins/WebHook/External/httplib.h:449]: This looks okay, though hopefully we'll just update the whole lib eventually as the newer version supports poll().

File: Plugins/SQL/Targets/SQLite.cpp [hunk 2]
```diff
@@ -92,17 +92,35 @@ std::optional<ResultSet> SQLite::ExecuteQuery()
 
     for (unsigned int i = 0; i < m_paramCount; i++)
     {
-        LOG_DEBUG("Binding value '%s' to param '%u'", m_paramValues[i], i);
-        // Params in SQLite are 1 based
-        int bindStatus = sqlite3_bind_text(m_stmt, i + 1, m_paramValues[i].c_str(), -1, nullptr);
+        if (m_paramValues[i])
+        {
+            LOG_DEBUG("Binding value '%s' to param '%u'", *m_paramValues[i], i);
```
[the remark this answers is not in the document]
Comment (drake127) [reply Plugins/SQL/Targets/SQLite.cpp:97]: If we can send nullptr to SQLite, it's great. I wasn't sure and found sqlite_bind_null instead. Should I change it?

[the remark this answers is not in the document]
Comment (mtijanic) [reply Plugins/SQL/Targets/SQLite.cpp:97]: I can merge this and we can fix in a follow-up if you want. Either works for me.

[the remark this answers is not in the document]
Comment (drake127) [reply Plugins/SQL/Targets/SQLite.cpp]: Done, you can quickly check and merge if it looks fine to you.

**Pull request merged**